Driverless AI/build-your-own-custom-recipe-tutorial/build-your-own-custom-recipe-tutorial.md

```diff
@@ -425,162 +474,210 @@ Driverless AI has Scorer recipes for the following categories:
 
-4\. Copy and paste the code above for **implementing scoring metric**, below the `#Extend base class` section of code of your .py file. Your text editor should look similar to the page below:
+Next h2o4gpu's confusion matrices will be called `cms = daicx.confusion_matrices(enc_actual.ravel(), enc_predicted.ravel(), sample_weight=cm_weights)` to compute the confusion matrices for ROC analysis for all possible prediction thresholds. We will then get the rows for the optimal metric `f1`, which you will see in the next section is the optimizer defined by variable `_threshold_optimizer = f1`. Then we will use cms to get the true negatives(tn), false positives(fp), false negative(fn) and true positives(tp) from those rows.
 
-![scorer-implementing-scoring-metric](assets/scorer-implementing-scoring-metric.jpg)
+Afterward the prediction thresholds for `tp`, `fp`, `tn` and `fn` are passed from the `cms` confusion matrices to the `protected_metric` method inside the lambda function, which is then passed to the `_metric` method where the **False Positive Rate** of `fp/(fp + tp)` is computed as long as the sum of `fp + tp` does not equal `0`. If that sum does equal zero (meaning division by zero) then the `_metric` method returns a `1`.Thus, the `_metric` method returns the **False Positive Rate** to the `protected_metric` method, which returns it to the lambda function, which assigns it to `cms['metric']`.  
+
+Finally, this `score` method will return the mean float score via `return cms['metric'].mean()`. This is the number we are trying to optimize, which can be a high or low value.
 
 ### Optimizing the Scorer
 
-In this final section, we will define for Driverless AI what makes a good scorer and when to use it. 
+Let's define for Driverless AI what makes a good scorer and when to use it. 
+
+7\. Copy and paste the following code below the `class MyFDRScorer(CustomScorer):` section of code in your .py file
 
 The python code for **optimizing the scorer**
 
 ~~~python
-#Optimizing the scorer
-   _threshold = 0.5
-   _binary = True
-   _regression = False
-   _multiclass = False
-   _maximize = False 
-   _perfect_score = 0
+_binary = True
+_multiclass = True
+_maximize = False
+_threshold_optimizer = "f1"
+_is_for_user = False  # don't let the user pick since can be trivially optimized (even when using F1-optimal thresh)
 ~~~
 
-**Threshold for binary predictions**
+Your text editor should look similar to the page below:
 
-Looking at the code above, we see the default value for the `_threshold,` which for this tutorial was set to 0.5 and can be adjusted. As mentioned in the previous section, this threshold will help determine set the values of 0 or 1 for the **actual** and **predicted**.
+![false-discovery-rate-scorer](assets/false-discovery-rate-scorer.jpg)
```

Comment:
@james94 I think it will be best to make this code copyable so that users can test the write code.

Reply:
Okay that is a good idea. I will also add the copyable code.